java/arcs/core/host/ParticleContext.kt

```diff
@@ -130,12 +138,13 @@ class ParticleContext(
      */
     suspend fun runParticle(notifyReady: (Particle) -> Unit) {
         withContext(requireNotNull(dispatcher)) {
-            check(particleState == ParticleState.Waiting) {
+            check(particleState in arrayOf(ParticleState.Waiting, ParticleState.Running)) {
```

Comment:
This line of code looks confusing to me as the err msg says "... after a successful call to initParticle" but you are using ParticleState to see if the check fails.

What about introducing a new state variable called "initParticleCalled"?

Reply:
I could do that. I was trying to keep this change as simple as possible while fixing the issues I'm encountering.

Reply:
(That is to say... I agree, but I don't want to block this PR on this sort of detali)

Reply:
Maybe leave a TODO?

Reply:
TODONE!